integration-tests/tests/helpers/client.rs

```diff
@@ -0,0 +1,67 @@
+use eyre::Result;
+use starknet::core::types::{BlockId, BlockTag, FieldElement, FunctionCall};
+use starknet::providers::jsonrpc::{HttpTransport, JsonRpcClient};
+use starknet::{macros::selector, providers::Provider};
+use std::error::Error;
+
+pub const EXECUTE: FieldElement = selector!("execute");
+
+pub struct KakarotClient<StarknetClient>
+where
+    StarknetClient: Provider,
+{
+    pub starknet_provider: StarknetClient,
+    pub kakarot_address: FieldElement,
+    pub proxy_account_class_hash: FieldElement,
+}
+
+impl KakarotClient<JsonRpcClient<HttpTransport>> {
+    pub fn new(starknet_provider: JsonRpcClient<HttpTransport>, kakarot_address: FieldElement, proxy_account_class_hash:FieldElement) -> Self {
+        Self {
+            starknet_provider,
+            kakarot_address,
+            proxy_account_class_hash,
+        }
+    }
+
+    pub async fn call_execute(
+        &self,
+        value: FieldElement,
+        bytecode: Vec<FieldElement>,
+        kakarot_calldata: Vec<FieldElement>,
+    ) -> Result<Vec<FieldElement>, Box<dyn Error>> {
+        let kakarot_calldata_len = FieldElement::from(kakarot_calldata.len());
+        let bytecode_len = FieldElement::from(bytecode.len());
+
+        let mut execute_arguments = vec![];
+        execute_arguments.push(value); //value
+
+        execute_arguments.push(bytecode_len); //bytecode_len
+        for bytecode_bytes in bytecode {
```

Comment:
in this case could be "bytecode_byte" as each element in the vector would be 1 byte and not several

Reply:
nice catch! corrected :)